core/core_kernel.cc

```diff
@@ -29,3 +29,85 @@ long long execute_kernel_busy_wait(const Kernel &kernel)
   }
   return acc;
 }
+
+// -- add by Yuankun
+void execute_kernel_memory(const Kernel &kernel)
+{
+	//currently use two src input
+	long long jump = kernel.kernel_arg.jump;
+	long long N = kernel.kernel_arg.input_bytes_per_src[0] / (sizeof(double)*2);
+	double *A = reinterpret_cast<double *>(kernel.kernel_arg.input_data[0]);
+	double *B = reinterpret_cast<double *>(kernel.kernel_arg.input_data[0] + N * sizeof(double));
+	double *C = reinterpret_cast<double *>(kernel.kernel_arg.output_data);
+
+	if( (kernel.kernel_arg.num_src_input==2) && (kernel.kernel_arg.input_data[1] != NULL) ){
+		N = N * 2;
+		B = reinterpret_cast<double *>(kernel.kernel_arg.input_data[1]);
+		// printf("B get 1\n");
+	}
```

Comment:
2 problems:
1. what if num_src_input == 3? shouldn't it be >= 2?
2. With your code, if num_src_input == 2, then total memory operations is 2N, but if num_src_input ==1, the total memory operations is N, then the task of num_src_input=1 is not the same with the num_src_input=2. You should always make sure all tasks are the same.